### PR TITLE
validators: implement validators API call

### PR DESCRIPTION
**Pull request by MatusKysel**

### CODE REVIEW

File: api/api_handler.go [hunk 1]
```diff
@@ -111,20 +111,26 @@ func (a *ApiHandler) GetAddressStats(ctx echo.Context, address AddressFilter) er
...
+
 	return ctx.JSON(http.StatusOK, response)
 }
 
 // GetValidatorsTotal returns total number of PBFT blocks produced in selected week
 func (a *ApiHandler) GetValidatorsTotal(ctx echo.Context, params GetValidatorsTotalParams) error {
+	stats := a.storage.GetWeekStats(int(params.Filter.Year), int(params.Filter.Week))
 	var count CountResponse
-	count.Total = 0
+	count.Total = uint64(stats.Total)
```
Comment (kstdl) [inline api/api_handler.go:133]: @rattrap do we need this data somewhere separate from that validators list?
If not I think it makes sense to merge this 2 requests

File: internal/indexer/block_context.go [hunk 2]
```diff
@@ -25,6 +27,9 @@ func MakeBlockContext(s *storage.Storage, client *chain.WsClient) *blockContext
 	bc.batch = s.NewBatch()
 	bc.client = client
 	bc.addressStats = make(map[string]*storage.AddressStats)
+	tn := time.Unix(int64(bc.age), 0)
+	year, week := tn.ISOWeek()
+	bc.WeekStats = s.GetWeekStats(year, week)
```
Comment (kstdl) [inline internal/indexer/block_context.go:32]: I think we can move this to some new function that would be called from process function and would be run in a goroutine
```
func UpdateValidatorStats() {
	tn := time.Unix(int64(bc.age), 0)
	year, week := tn.ISOWeek()
	weekStats := s.GetWeekStats(year, week)
    weekStats.AddPbftBlock(block)
	bc.batch.UpdateWeekStats(weekStats)
}
```

**Review by kstdl — APPROVED** [review]

File: api/api_handler.go (hunk 1 above)
Comment (rattrap) [reply api/api_handler.go:133]: Yes. We have a summary at the start of the page - https://testnet.explorer.taraxa.io/node

And then the list of validators under that. "Total" for the validators list may mean total number of block for that page or total number of validators returned. At the top of the page we need the total for all validators (all pages) for the week.

Comment (kstdl) [reply api/api_handler.go:133]: yeah, but if it is on the same page we can get all the data with 1 request

Comment (rattrap) [reply api/api_handler.go:133]: We could have it in the validators request too and when integrating we can have to option to just use one request.

**Pull request merged**